common/src/main/java/org/conscrypt/TrustManagerImpl.java

```diff
@@ -970,52 +970,55 @@ private TrustAnchor findTrustAnchorBySubjectAndPublicKey(X509Certificate cert) {
      * {@code null} (the default), endpoint identification will use the default hostname verifier
      * set in {@link HttpsURLConnection#setDefaultHostnameVerifier(javax.net.ssl.HostnameVerifier)}.
      */
-    synchronized static void setDefaultHostnameVerifier(HostnameVerifier verifier) {
+    synchronized static void setDefaultHostnameVerifier(ConscryptHostnameVerifier verifier) {
         defaultHostnameVerifier = verifier;
     }
 
     /**
      * Returns the currently-set default hostname verifier.
      *
-     * @see #setDefaultHostnameVerifier(HostnameVerifier)
+     * @see #setDefaultHostnameVerifier(ConscryptHostnameVerifier)
      */
-    synchronized static HostnameVerifier getDefaultHostnameVerifier() {
+    synchronized static ConscryptHostnameVerifier getDefaultHostnameVerifier() {
         return defaultHostnameVerifier;
     }
 
     /**
      * Set the hostname verifier that will be used for HTTPS endpoint identification.  If
      * {@code null} (the default), endpoint identification will use the default hostname verifier
-     * set in {@link #setDefaultHostnameVerifier(HostnameVerifier)}.
+     * set in {@link #setDefaultHostnameVerifier(ConscryptHostnameVerifier)}.
      */
-    void setHostnameVerifier(HostnameVerifier verifier) {
+    void setHostnameVerifier(ConscryptHostnameVerifier verifier) {
         this.hostnameVerifier = verifier;
     }
 
     /**
      * Returns the currently-set hostname verifier for this instance.
      *
-     * @see #setHostnameVerifier(HostnameVerifier)
+     * @see #setHostnameVerifier(ConscryptHostnameVerifier)
      */
-    HostnameVerifier getHostnameVerifier() {
+    ConscryptHostnameVerifier getHostnameVerifier() {
         return hostnameVerifier;
     }
 
-    private HostnameVerifier getHttpsVerifier() {
+    private enum GlobalHostnameVerifierAdapter implements ConscryptHostnameVerifier {
+        INSTANCE;
+
+        @Override
+        public boolean verify(String hostname, SSLSession session) {
+            return HttpsURLConnection.getDefaultHostnameVerifier().verify(hostname, session);
+        }
+    }
+
+    private ConscryptHostnameVerifier getHttpsVerifier() {
         if (hostnameVerifier != null) {
             return hostnameVerifier;
         }
-        HostnameVerifier defaultVerifier = getDefaultHostnameVerifier();
+        ConscryptHostnameVerifier defaultVerifier = getDefaultHostnameVerifier();
```

Comment:
Up to you, but would it not make sense to just initialise TrustManagerImpl.defaultHostnameManager to ConscryptHostnameVerifier.INSTANCE?

Reply:
I think we're going to keep it this way because initializing the default that way makes the setter need logic to map null to ConscryptHostnameVerifier.INSTANCE, which I think is slightly less clear than this.  I think it's pretty close, though.